server/routes/chat.go

```diff
@@ -0,0 +1,147 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package routes
+
+import (
+	"encoding/json"
+	"fmt"
+
+	//"io/ioutil"
+	"localdev/main/models"
+	"localdev/main/services"
+	"log"
+	"net/http"
+
+	"cloud.google.com/go/datastore"
+	"github.com/gorilla/mux"
+	"google.golang.org/api/iterator"
+)
+
+func CheckChatDatastore(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	query := datastore.NewQuery("Chat")
+	it := services.Locator.DsClient().Run(ctx, query)
+	for {
+		var chat_room models.ChatRoom
+		_, err := it.Next(&chat_room)
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			log.Printf("Error fetching next chatroom: %v", err)
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+		fmt.Fprintf(w, "ChatRoom %q, Messages %q\n", chat_room.ID, chat_room.Messages)
+	}
+	w.Write([]byte("done"))
+}
+
+func GetMessagesFromChatRoom(w http.ResponseWriter, r *http.Request) {
+	if r.Method != http.MethodGet {
+		log.Printf("Invalid request method")
+		w.WriteHeader(http.StatusMethodNotAllowed)
+		w.Write([]byte("Invalid request method"))
+		return
+	}
+
+	ctx := r.Context()
```

Comment:
This should be `ctx := appengine.NextContext(r)` in order for `userService.GetCurrentUserID(ctx)` to work correctly when deployed

Reply:
Are you sure is NextContext? I cannot find this function in "google.golang.org/appengine". Maybe NewContext(r)?

Reply:
Yep, guess I mistyped. I meant NewContext